src/lib/autoMode.js

```diff
@@ -0,0 +1,67 @@
+  `dt=0.01&fo=0.998&dp=0.009&cm=3&cx=-1.6545999999999994&cy=-0.3642500000000002&w=24.7354&h=24.7354&pc=3000&vf=%2F%2F p.x and p.y are current coordinates%0A%2F%2F v.x and v.y is a velocity at point p%0Avec2 get_velocity(vec2 p) {%0A  vec2 v %3D vec2(0.%2C 0.)%3B%0A%0A  %2F%2F change this to get a new vector field%0A  float dt %3D 0.01%3B%0Afloat t %3D frame*dt%3B%0Afloat w %3D 2.*PI%2F5.%3B%0Afloat A %3D 2.%3B%0A%0Afloat d %3D sqrt(p.x*p.x %2B p.y*p.y)%3B%0Av.x %3D A*cos(w*t%2Fd)%3B%0Av.y %3D A*sin(w*t%2Fd)%3B%0A%0A  return v%3B%0A}`,
+  `dt=0.01&fo=0.998&dp=0.009&cm=2&cx=-1.424&cy=0.92285&w=8.541&h=8.541&vf=%2F%2F p.x and p.y are current coordinates%0A%2F%2F v.x and v.y is a velocity at point p%0Avec2 get_velocity(vec2 p) {%0A  vec2 v %3D vec2(0.%2C 0.)%3B%0A%0A  %2F%2F change this to get a new vector field%0A  float size %3D 1.0%3B%0Avec2 o %3D (p)%2F(1.6* size)%3B%0A  float a %3D o.x*o.x%2Bo.y*o.y-0.3%3B%0A v %3D vec2(step(a*a*a%2C o.x*o.x*o.y*o.y*o.y))%3B%0A  %0A%0A  return v%3B%0A}`
+];
+
+let delayTime, incoming;
+
+export function initAutoMode() {
+  const auto = appState.getQS().get('auto');
+  if (!auto) {
+    return;
+  }
+
+  if (!/s$/i.test(auto)) {
+    console.error('malformed auto param; missing seconds specifier');
+    return;
+  }
+
+  let seconds = parseFloat(auto.replace(/s$/i, ''));
+  if (Number.isNaN(seconds)) {
+    console.error('malformed auto param; not a number');
+    return;
+  }
+
+  if (seconds <= 1) {
+    console.warn('auto param seconds too small; defaulting to 30');
+    seconds = 30;
+  }
+
+  delayTime = seconds * 1000;
+
+  setTimeout(next, delayTime);
+}
+
+function next() {
+  if (!incoming || !incoming.length) {
+    // TODO: shuffle
+    incoming = presets.slice();
+  }
+
+  const presetString = incoming.shift();
+  const parts = presetString.split('&');
```

Comment:
This parsing can be removed if JSON objects are used as presets.

Reply:
Indeed, though it would still be nice to easily convert from URLs to JSON objects... I've made a util file for that.